core/corehttp/gateway_handler.go

```diff
@@ -347,11 +347,17 @@ func (i *gatewayHandler) getOrHeadHandler(ctx context.Context, w http.ResponseWr
 		}
 	}
 
+	var hash string
+	if !strings.Contains(originalUrlPath, "ipfs") {
```

Comment:
We should use HasPrefix with `ipfsPathPrefix`. This won't catch IPNS links with the word `ipfs` somewhere in the path.